db/migrate/20180710172959_disallow_multi_levels_theme_components.rb

```diff
@@ -0,0 +1,68 @@
+class DisallowMultiLevelsThemeComponents < ActiveRecord::Migration[5.2]
```

Comment:
The migration is a little too complicated for me to understand without being able to see the actual data moving. @OsamaSayegh What is the worst possible outcome if this migration goes wrong?

Reply:
worst possible outcome is a mess in theme dependencies, or a migration that refused to run

Reply:
> worst possible outcome is a mess in theme dependencies

Would we be able to migrate the data in such a way that we can still recover if the dependencies become a mess? If not, are we able to break the migration into smaller steps? I'm not sure what is possible but my sense is that this migration is tricky one.

Reply:
It's a tricky one indeed. I can provide you with before and after data for a few themes so that you can understand what this migration does and see if the data looks correct.